### ISSUE TITLE
aws_ec2_instance_type_offerings should have an attribute for location_types and locations

### ISSUE THREAD

**Issue by robomon1** [hashicorp/terraform-provider-aws#15272]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version
Terraform v0.13.3
+ provider registry.terraform.io/-/aws v3.7.0
+ provider registry.terraform.io/hashicorp/aws v3.7.0

### Affected Resource(s)

aws_ec2_instance_type_offerings

### Terraform Configuration Files

```hcl
terraform {
  required_version = "~> 0.13.3"
}

provider "aws" {
  region = "us-west-2"
}

# Get all available availability zones
data "aws_availability_zones" "available" {
  state = "available"
}

output "zone_names" {
  value = data.aws_availability_zones.available.names
}
output "zone_ids" {
  value = data.aws_availability_zones.available.zone_ids
}

# We want only the availability zones where t2.micro is supported
data "aws_ec2_instance_type_offerings" "az" {
  filter {
    name   = "instance-type"
    values = ["t2.micro"]
  }

  location_type = "availability-zone"
}

# We can get instance_types attribute
output "location_types" {
  value = data.aws_ec2_instance_type_offerings.az.instance_types
}

# But what we really want is a locations attribute which gives me
# just the locations supported by the instance types as shown by the
# aws cli for describe-instance-type-offerings.  There are 4 availability
# zones but ony 3 of them support t2.micro

# aws ec2 describe-instance-type-offerings --location-type availability-zone \
#            --filters Name=instance-type,Values=t2.micro \
#            --region us-west-2
# {
#     "InstanceTypeOfferings": [
#         {
#             "InstanceType": "t2.micro",
#             "LocationType": "availability-zone",
#             "Location": "us-west-2b"
#         },
#         {
#             "InstanceType": "t2.micro",
#             "LocationType": "availability-zone",
#             "Location": "us-west-2a"
#         },
#         {
#             "InstanceType": "t2.micro",
#             "LocationType": "availability-zone",
#             "Location": "us-west-2c"
#         }
#     ]
# }
```

### Expected Behavior

I expect to get all the attributes including LocationTypes and Locations from the datasource

### Actual Behavior

There is no way to get to the LocationTypes and Locations data returned by an aws_ec2_instance_type_offering datasource.

### Steps to Reproduce

1. `terraform init`
2. `terraform apply`

### Important Factoids

I'm using us-west-2 because the t2.micro is only supported in us-west-2a, us-west-2b, and us-west-2c but there is also a us-west-2d az

**Comment by mjulianotq** [comment]
#10989 is where the current functionality was introduced. (Just connecting related data.)

**Comment by mjulianotq** [comment]
I've submitted PR #17794 which adds these fields to the data. Please go give it a thumbs up if you want this feature.

I have need of the feature to dynamically choose which availability zones are used in a cluster based on the instance type chosen for the cluster. Right now, I'm just crossing my fingers that the first 2 randomly chosen AZs support my chosen instance type, and if they don't, deleting and trying again. The instance type I've been testing with is supported in all but one of the AZs in us-east-1, which has been fine for testing, but I'm nearing production on my module, so I need this badly enough to do it myself.

**Comment by ewbankkit** [comment]
Closes via #16704.
